Mathlib/CategoryTheory/Bicategory/LaxSlice.lean

```diff
@@ -0,0 +1,526 @@
+|          |          |
+|  ⇒⇒θ₀⇒⇒  f₁ ⇒⇒θ₁⇒⇒  |
+|          ↓          |
+|----f₀--->X<---f₂----|
+``` -/
+@[simps]
+def comp₁ {A B C : F ↓ X} (p₀ : Hom₁ F X A B) (p₁ : Hom₁ F X B C) : Hom₁ F X A C where
+  dom_map := p₀.dom_map ≫ p₁.dom_map
+  cod_map := p₀.cod_map ≫ (F.map p₀.dom_map ◁ p₁.cod_map)
+    ≫ (α_ (F.map p₀.dom_map) (F.map p₁.dom_map) C.map).inv
+    ≫ (F.mapComp p₀.dom_map p₁.dom_map ▷ C.map)
+
+/-- Underlying CategoryStruct on objects. -/
+@[simps]
+instance : CategoryStruct (F ↓ X) where
+  Hom A B := Hom₁ F X A B
+  id A := id₁ F X A
+  comp f g := comp₁ F X f g
+
+/-- 2-cells in `F ↓ X`
+A 2-cell `(p₀, θ₀) ⟶ (p₁, θ₁)` is a 2-cell `α : p₀ ⟶ p₁` in `B` such that
+`Fα` satisfies the ice cream cone condition:
+```
+|-------Fp₁-----|     FA₀-----Fp₁---->FA₁
+|        ⇑      |     |               |
+|       Fα      |     |               |
+|        ⇑      ↓     |               |
+FA₀-----Fp₀---->FA₁ = |     ⇒⇒θ₁⇒⇒    |
+|               |     |               |
+|     ⇒⇒θ₀⇒⇒    |     |               |
+|               |     |               |
+|--f₀-->X<--f₁--|     |--f₀-->X<--f₁--|
+``` -/
+@[ext]
+structure Hom₂ {A B : F ↓ X} (f : A ⟶ B) (g : A ⟶ B) where
+  map : f.dom_map ⟶ g.dom_map
+  icc : f.cod_map ≫ (F.map₂ map ▷ B.map) = g.cod_map
+
+/-- Identity 2-cell.
+For a 1-cell `(p, θ)`, the identity 2-cell is `1_p` -/
+@[simps]
+def id₂ {A B : F ↓ X} (f : A ⟶ B) : Hom₂ F X f f where
+  map := 𝟙 f.dom_map
+  icc := by simp
+
+/-- Vertical composition of 2-cells.
+For 1-cells `(p, θ), (p', θ'), (p'', θ'') : (A₀, F₀) ⟶ (A₁, F₁)`
+and 2-cells `α : (p, θ) ⟶ (p', θ'), α' : (p', θ') ⟶ (p'', θ'')`,
+their vertical composite is the composite `α'α : (p, θ) ⟶ (p'', θ'')`. -/
+@[simps]
+def comp₂ {A B : F ↓ X} {f g h : A ⟶ B} (α : Hom₂ F X f g) (β : Hom₂ F X g h) :
+    Hom₂ F X f h where
+  map := α.map ≫ β.map
+  icc := by simp [←α.icc, ←β.icc]
+
+/-- Category structure on 1-cells with vertical composition. -/
+instance (A B : F ↓ X) : Category (A ⟶ B) where
+  Hom f g := Hom₂ F X f g
+  id f := id₂ F X f
+  comp η θ := comp₂ F X η θ
+
+/-- Whisker a 2-cell on the left by a 1-cell.
+Comes precisely from the whiskering on `B`. -/
+@[simps]
+def whiskerLeft {A B C : F ↓ X} (f : A ⟶ B) {g h : B ⟶ C} (η : g ⟶ h) :
+    (f ≫ g) ⟶ (f ≫ h) where
+  map := f.dom_map ◁ η.map
+  icc := by simp only [comp_def, comp₁_dom_map, comp₁_cod_map, assoc,
+                       ← comp_whiskerRight, LaxFunctor.mapComp_naturality_right,
+                       ← η.icc, whiskerLeft_comp]
+            simp
+
+@[simp]
+theorem whiskerLeft_id {A B C : F ↓ X} (f : A ⟶ B) (g : B ⟶ C) :
+    whiskerLeft F X f (𝟙 g) = 𝟙 (f ≫ g) := by
+  refine Hom₂.ext ?_
+  change f.dom_map ◁ 𝟙 g.dom_map = 𝟙 _
+  simp
+
+@[simp]
+theorem whiskerLeft_comp {A B C : F ↓ X} (f : A ⟶ B) {g h i : B ⟶ C} (η : g ⟶ h) (θ : h ⟶ i) :
+    whiskerLeft F X f (η ≫ θ) = whiskerLeft F X f η ≫ whiskerLeft F X f θ := by
+  refine Hom₂.ext ?_
+  change f.dom_map ◁ (η.map ≫ θ.map) = _ ≫ _
```

Comment:
All these `change` statements suggest that you don't quite have the correct simp set. Preferably `simp` should just take care of this